src/imitation/rewards/discrim_net.py

```diff
@@ -277,89 +212,98 @@ def __init__(self, reward_net: reward_net.RewardNet, entropy_weight: float = 1.0
+            state, action, next_state, done
+        )
+        return log_policy_act_prob - reward_output_train
 
-    @property
-    def next_obs_ph(self):
-        return self.reward_net.next_obs_ph
+    def reward_test(
+        self,
+        state: th.Tensor,
+        action: th.Tensor,
+        next_state: th.Tensor,
+        done: th.Tensor,
+    ) -> th.Tensor:
+        rew = self.reward_net._reward_test(state, action, next_state, done)
+        assert rew.shape == state.shape[:1]
+        return rew
 
-    @property
-    def done_ph(self):
-        return self.reward_net.done_ph
+    def reward_train(
+        self,
+        state: th.Tensor,
+        action: th.Tensor,
+        next_state: th.Tensor,
+        done: th.Tensor,
+    ) -> th.Tensor:
+        """Compute train reward. This reward does *not* include an entropy
+        bonus; the entropy bonus should be added directly to PPO, SAC, etc."""
+        rew = self.reward_net._reward_train(state, action, next_state, done)
+        assert rew.shape == state.shape[:1]
+        return rew
 
-    def build_graph(self):
-        self._disc_logits_gen_is_high = (
-            self.log_policy_act_prob_ph - self.reward_net.reward_output_train
-        )
 
-        # Construct discriminator loss.
-        self._disc_loss = tf.nn.sigmoid_cross_entropy_with_logits(
-            labels=tf.cast(self.labels_gen_is_one_ph, tf.float32),
-            logits=self._disc_logits_gen_is_high,
-            name="disc_loss",
-        )  # (None,)
-
-        # Construct generator reward:
-        # \[\hat{r}(s,a) = \log(D_{\theta}(s,a)) - \log(1 - D_{\theta}(s,a)).\]
-        # This simplifies to:
-        # \[\hat{r}(s,a) = f_{\theta}(s,a) - \log \pi(a \mid s).\]
-        # This is just an entropy-regularized objective.
-
-        ent_bonus = -self.entropy_weight * self.log_policy_act_prob_ph
-        self._policy_train_reward = self.reward_net.reward_output_train + ent_bonus
-        self._policy_test_reward = self.reward_net.reward_output_test
-
-        self.reward_net.build_summaries()
-
-    def _save(self, directory):
-        os.makedirs(directory, exist_ok=True)
-        params = {
-            "entropy_weight": self.entropy_weight,
-            "reward_net_cls": type(self.reward_net),
-        }
-        with open(os.path.join(directory, "args"), "wb") as f:
-            pickle.dump(params, f)
-        return self.reward_net.save(os.path.join(directory, "reward_net"))
-
-    @classmethod
-    def _load(cls, directory):
-        with open(os.path.join(directory, "args"), "rb") as f:
-            params = pickle.load(f)
-        reward_net_cls = params.pop("reward_net_cls")
-        reward_net = reward_net_cls.load(os.path.join(directory, "reward_net"))
-        return cls(reward_net=reward_net, **params)
-
-
-DiscrimNetBuilder = Callable[
-    [Sequence[tf.Tensor]], Tuple[tf.Tensor, networks.LayersDict]
-]
+DiscrimNetBuilder = Callable[..., nn.Module]
```

Comment:
Where is this used? I expect that it can be removed if I move to passing fully-constructed discriminators into these classes.